### PR TITLE
Add VectorUtil

### PR DESCRIPTION
**Pull request by Simplicitee**
### Additions ➕ 
* Adds VectorUtil class
    * Focuses on implementing Vector functionality that is either complex or not present within Bukkit
* Adds Angle class
    * Allows for easier use of angles within the project without requiring developers to give angle parameters in specific units
* Adds UnitVector enum
    * Allows for easy use of unit vectors in a nice format

## Checklist
- [x] I have tested my changes to the best of my abilities.
- [x] I have linked to any related GitHub Issues or Trello cards.
- [x] I have made sure my code is clean, well documented, and matches the pre-existing style of the codebase.
- [x] I understand that my pull request will only be merged after it is reviewed by maintainers and any requested changes are made.

### CODE REVIEW

File: src/main/java/com/projectkorra/core/math/Angle.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+		return new Angle(AngleMode.DEGREES, (double) value);
+	}
+
+	public static Angle degrees(float value) {
+		return new Angle(AngleMode.DEGREES, (double) value);
+	}
+
+	public static Angle degrees(double value) {
+		return new Angle(AngleMode.DEGREES, value);
+	}
```
Comment (ChristopherWMM) [inline src/main/java/com/projectkorra/core/math/Angle.java]: Java is able to implicitly cast `int` and `float` to `double` so I don't think you need anything other than the double typed methods.

File: src/main/java/com/projectkorra/core/util/VectorUtil.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+	}
+	
+	/**
+	 * Returns an orthogonal vector from an axis
+	 * @param axis vector to be orthogonal of
+	 * @param length magnitude of orthogonal vector
+	 * @param rotation counterclockwise rotation, 0 = righthand side of the axis
+	 * @return an orthogonal vector
+	 */
+	public static Vector orthogonal(Vector axis, double length, Angle rotation) {
```
Comment (ChristopherWMM) [inline src/main/java/com/projectkorra/core/util/VectorUtil.java]: I would prefer that you add the `throws IllegalArgumentException` declaration to the method header that could be caused from the `Validate.isTrue(...)` call. Additionally, it should go in the Javadocs.

File: src/main/java/com/projectkorra/core/util/VectorUtil.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+		double yaw = Math.toRadians(getYaw(axis));
+		Vector other = new Vector(-Math.sin(yaw), axis.getY() - 1, Math.cos(yaw));
+		Vector third;
+		
+		if (getPitch(other) > getPitch(axis)) {
+			third = other.getCrossProduct(axis);
+		} else {
+			third = axis.getCrossProduct(other);
+		}
```
Comment (ChristopherWMM) [inline src/main/java/com/projectkorra/core/util/VectorUtil.java]: Maybe convert to a ternary assignment?

File: src/main/java/com/projectkorra/core/math/Angle.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
+package com.projectkorra.core.math;
```
Comment (ChristopherWMM) [inline src/main/java/com/projectkorra/core/math/Angle.java]: I think it makes more sense to have this new math directory to be a sub-directory of `util` as opposed to the root source directory.

**Review by ChristopherWMM — CHANGES_REQUESTED** [review]

**Review by v-h-gn — APPROVED** [review]
Everything looks good, I can't say I'd vouch for efficiency but I think this is a good base.

File: src/main/java/com/projectkorra/core/util/VectorUtil.java [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+		    return vector.getY() > 0 ? -90 : 90;
+		}
+		
+		double xz = Math.sqrt(x * x + z * z);
+		return (float) Math.toDegrees(Math.atan(-vector.getY() / xz));
+	}
+	
+	/**
+	 * Gets the yaw of the given vector, a vector pointing straight 
+	 * up or down will return pi/2 or -pi/2 respectively
```
Comment (ChristopherWMM) [inline src/main/java/com/projectkorra/core/util/VectorUtil.java]: This line needs to be changed to properly reflect that the method returns the degree value and not the radian value.

**Review by ChristopherWMM — APPROVED** [review]
Looks good :+1:

**Pull request merged**